code/xcvm/lib/proto/src/lib.rs

```diff
@@ -0,0 +1,452 @@
+				xcvm_core::BindingValue::Asset(asset_id.try_into()?),
+		})
+	}
+}
+
+impl<Account> TryFrom<Spawn> for XCVMInstruction<Account>
+where
+	Account: From<Vec<u8>>,
+{
+	type Error = ();
+
+	fn try_from(spawn: Spawn) -> core::result::Result<Self, Self::Error> {
+		let network = spawn.network.ok_or(())?.into();
+		let salt = spawn.salt.ok_or(())?.salt;
+		// let program = spawn.program.ok_or(())?;
+		// TODO(aeryz): convert program
+		// TODO(aeryz): assets conversion can be a function
+		Ok(XCVMInstruction::Spawn {
+			network,
+			salt,
+			bridge_security: match spawn.security {
+				0 => xcvm_core::BridgeSecurity::None,
+				1 => xcvm_core::BridgeSecurity::Deterministic,
+				2 => xcvm_core::BridgeSecurity::Probabilistic,
+				3 => xcvm_core::BridgeSecurity::Optimistic,
+				_ => return Err(()),
+			},
+			assets: Funds(
+				spawn
+					.assets
+					.into_iter()
+					.map(|asset| asset.try_into())
+					.collect::<core::result::Result<Vec<_>, _>>()?,
+			),
+			program: XCVMProgram { tag: Vec::new(), instructions: VecDeque::new() },
+		})
+	}
+}
+
+impl From<Network> for NetworkId {
+	fn from(network: Network) -> Self {
+		(network.network_id as u8).into()
+	}
+}
+
+impl<Account> TryFrom<Transfer> for XCVMInstruction<Account>
+where
+	Account: From<Vec<u8>>,
+{
+	type Error = ();
+
+	fn try_from(transfer: Transfer) -> core::result::Result<Self, Self::Error> {
+		let account_type = transfer.account_type.ok_or(())?;
+		Ok(XCVMInstruction::Transfer {
+			to: account_type.into(),
+			assets: Funds(
+				transfer
+					.assets
+					.into_iter()
+					.map(|asset| asset.try_into())
+					.collect::<core::result::Result<Vec<_>, _>>()?,
+			),
+		})
+	}
+}
+
+impl<Acc> From<transfer::AccountType> for Destination<Acc>
+where
+	Acc: From<Vec<u8>>,
+{
+	fn from(account_type: transfer::AccountType) -> Self {
+		match account_type {
+			transfer::AccountType::Account(Account { account }) =>
+				Destination::Account(account.into()),
+			transfer::AccountType::Relayer(_) => Destination::Relayer,
+		}
+	}
+}
+
+impl TryFrom<Asset> for (xcvm_core::AssetId, Amount) {
+	type Error = ();
+
+	fn try_from(asset: Asset) -> core::result::Result<Self, Self::Error> {
+		let asset_id = asset.asset_id.ok_or(())?.try_into()?;
+		let amount = asset.balance.ok_or(())?.try_into()?;
+
+		Ok((asset_id, amount))
+	}
+}
+
+impl TryFrom<AssetId> for xcvm_core::AssetId {
+	type Error = ();
+
+	fn try_from(asset_id: AssetId) -> core::result::Result<Self, Self::Error> {
+		Ok(xcvm_core::AssetId(asset_id.asset_id.ok_or(())?.into()))
+	}
+}
+
+impl TryFrom<Balance> for Amount {
+	type Error = ();
+
+	fn try_from(balance: Balance) -> core::result::Result<Self, Self::Error> {
+		let balance_type = balance.balance_type.ok_or(())?;
+		let wrap = |num: u128| -> FixedU128<U16> { FixedU128::wrapping_from_num(num) };
+		// TODO(aeryz): This can be a helper function in SDK so that users won't
+		// necesarilly need to know how the ratio is handled in our SDK
```

Comment:
```suggestion
		// necessarily need to know how the ratio is handled in our SDK
```
Nit typo, failing spellcheck CI